src/averageAges.js

```diff
@@ -1,68 +1,87 @@
 'use strict';
-
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
+/*
  * @param {object[]} people
- * @param {number} century - optional
+ * @param {number} century
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const averageAgeMan = people
+    .filter((person) => century ? isMale(person.sex)
+    && Math.ceil(person.died / 100) === century : isMale(person.sex))
+    .map((getAge) => getAge.died - getAge.born);
+
+  const averagAge = averageAges(averageAgeMan);
+
+  return averagAge;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
  * @param {object[]} people
- * @param {boolean} withChildren - optional
+ * @param {boolean} withChildren
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const averageAgeWoman = people
+    .filter(({ sex, name }) => {
+      return withChildren ? isFemale(sex)
+      && people.some(child => child.mother === name)
+        : isFemale(sex);
+    })
+    .map((getAge) => getAge.died - getAge.born);
```

Comment:
let's keep the same syntax in the project. If you use decomposing here, do it in the previous function too. Also you can use shorthand without `return`